hypothesis-python/tests/cover/test_lookup.py

```diff
@@ -1053,3 +1058,38 @@ def f(x: int):
     msg = "@no_type_check decorator prevented Hypothesis from inferring a strategy"
     with pytest.raises(TypeError, match=msg):
         st.builds(f).example()
+
+
+class TupleSubtype(tuple):
+    pass
+
+
+def test_tuple_subclasses_not_generic_sequences():
+    # see https://github.com/HypothesisWorks/hypothesis/issues/3767.
+    with temp_registered(TupleSubtype, st.builds(TupleSubtype)):
+        s = st.from_type(typing.Sequence[int])
+        assert_no_examples(s, lambda x: isinstance(x, tuple))
+
+
+T = typing.TypeVar("T")
+
+
+@typing.runtime_checkable
+class Fooable(typing.Protocol[T]):
+    def foo(self):
+        ...
+
+
+class FooableConcrete(tuple):
+    def foo(self):
+        pass
+
+
+def test_only_tuple_subclasses_in_typing_type():
```

Comment:
I'd just skip this on Python 3.8, since it's using later typing features.  

If we could _also_ get a subclassing-based test working that'd make me more confident that future changes won't accidentally break something, but I'd be happy to merge without that too.

Reply:
I slept on it, but I'm drawing a blank on a subclassing-based test (as opposed to using `Protocol`) where the only valid types are tuple types. I'd be happy to add it if someone else can come up with one, though.